src/background.js

```diff
@@ -67,24 +67,32 @@ const onHeadersReceived = events.onHeadersReceived.bind(events);
 
 // Cliqz Modules
 const humanweb = cliqz.modules['human-web'];
-const { adblocker, antitracking } = cliqz.modules;
+const { adblocker, antitracking, hpn } = cliqz.modules;
 const messageCenter = cliqz.modules['message-center'];
 const offers = cliqz.modules['offers-v2'];
 
+log('MODULES', cliqz.modules);
+
+const CORRECT_STATE = 'CorrectState';
+/**
+ * Enable or disable specified module.
+ * @memberOf Background
+ * @param {Object} module  Cliqz module
+ * @param {boolean} enabled true - enable, false - disable
+ * @return {Promise}
+ */
 function setCliqzModuleEnabled(module, enabled) {
-	if (enabled) {
+	if (enabled && !module.isEnabled) {
+		log('SET CLIQZ MODULE ENABLED', module);
 		return cliqz.enableModule(module.name);
+	} else if (!enabled && module.isEnabled) {
+		log('SET CLIQZ MODULE DISABLED', module);
+		return module.isReady().then(() => cliqz.disableModule(module.name));
 	}
-	return module.isReady().then(() => cliqz.disableModule(module.name));
+	log('MODULE IS ALREADY IN CORRECT STATE', module, enabled);
+	return Promise.resolve(CORRECT_STATE);
```

Comment:
Let's add a few checks here to make sure the `module` parameter actually is indeed a fully qualified module object. Otherwise make the function fail gracefully.